src/components/progress-circle/progress-circle.ts

```diff
@@ -41,47 +84,172 @@ export class MdProgressCircle {
    * mode is bound to the host as the attribute host.
    */
   @HostBinding('attr.mode')
-  @Input() mode: 'determinate' | 'indeterminate' = 'determinate';
+  @Input()
+  get mode() {
+    return this._mode;
+  }
+  set mode(m: 'determinate' | 'indeterminate') {
+    if (m == 'indeterminate') {
+      this._startIndeterminateAnimation();
+    } else {
+      this._cleanupIndeterminateAnimation();
+    }
+    this._mode = m;
+  }
+  private _mode: 'determinate' | 'indeterminate' = 'determinate';
+
+
+  constructor(ref: ChangeDetectorRef) {
+    this._ref = ref;
+  }
 
 
   /**
-   * Gets the current stroke dash offset to represent the progress circle.
-   *
-   * The stroke dash offset specifies the distance between dashes in the circle's stroke.
-   * Setting the offset to a percentage of the total circumference of the circle, fills this
-   * percentage of the overall circumference of the circle.
+   * Animates the rendered circle.
    */
-  strokeDashOffset() {
-    // To determine how far the offset should be, we multiple the current percentage by the
-    // total circumference.
+  private _renderCircle(animateFrom: number, animateTo: number, ease: Function,
+                        duration: number, rotation: number) {
+    let id = ++this._lastAnimationId;
+    let startTime = MdProgressCircle._animationNow();
+    let changeInValue = animateTo - animateFrom;
+    let animationDuration = duration || 225;
+
+    // No need to animate it if the values are the same
+    if (animateTo === animateFrom) {
+      this.currentPath = MdProgressCircle.getSvgArc(animateTo, rotation);
+    } else {
+      let animation = (now: number) => {
+        let elapsedTime = Math.max(
+          0, Math.min((now || MdProgressCircle._animationNow()) - startTime, animationDuration));
 
-    // The total circumference is calculated based on the radius we use, 45.
-    // PI * 2 * 45
-    return 251.3274 * (100 - this._value) / 100;
+        this.currentPath = MdProgressCircle.getSvgArc(
+          ease(elapsedTime, animateFrom, changeInValue, animationDuration),
+          rotation
+        );
+
+        // Prevent overlapping animations by checking if a new animation has been called for and
+        // if the animation has lasted long than the animation duration.
+        if (id === this._lastAnimationId && elapsedTime < animationDuration) {
+          window.requestAnimationFrame(animation);
```

Comment:
You shouldn't need to use `window.` anywhere. Just `requestAnimationFrame` or `setInterval` etc.